### DISCUSSION TITLE
Why are XDG dirs disabled on macOS?

### DISCUSSION THREAD

**Discussion by steinuil** [abiosoft/colima#1016] [Q&A] [answered]
I noticed that even with `XDG_CONFIG_HOME` set in the environment, colima still uses `$HOME/.colima` as config directory on macOS. I traced it down to this condition, but the commit log doesn't offer any reasons that I could find for this change.

https://github.com/abiosoft/colima/blob/d20a53a71c380a6160d61a602b1dc9573359b224/config/dirs.go#L50-L51

What is `QEMU_SYSTEM_ARCH` and where upstream should it be handled, and why does it impact the placement of the config dir on macOS? Is it handled now?

**Comment by binary-signal** [comment]
this is not expected behavior, I hope it's a bug that slipped away

**Comment by abiosoft** [comment, marked as answer]
It has to do with how Colima was launching QEMU to override some flags that were not supported upstream in Lima.

But it indeed got slipped away, the issue has been resolved for a while.

**Reply by abiosoft** [reply]
This has been reinstated for macOS on the current development version and would be part of the next release.

```sh
# to try it now
brew install --head colima
```

**Reply by andrewcrook** [reply]
just a note, if you use `brew services` to start, the service's plist will not have the XDG variables by default so will drop a `~/.colima` directory when started from launchd. So when you go to use from command-line you will still get a warning about it ignoring XDG. To get around this have to add the XDG paths as environment variables to the users LaunchAgents plist file for the service. If you don't start colima this way you don't have to do anything.

**Reply by metametapod** [reply]
Can also be set via: `launchctl setenv XDG_CONFIG_HOME ~/.config`.
